modules/claims_api/app/controllers/claims_api/base_form_controller.rb

```diff
@@ -47,7 +47,28 @@ def auth_headers
 
     def documents
       document_keys = params.keys.select { |key| key.include? 'attachment' }
-      params.slice(*document_keys).values
+      params.slice(*document_keys).values.map do |document|
+        if document.is_a?(String)
+          decode_document(document)
+        else
+          document
+        end
+      end.compact
+    end
+
+    def decode_document(document)
+      match = document.match(/\;base64\,(.+?)$/)
+      if match
+        base64 = match.captures.first
+        decoded_data = Base64.decode64(base64)
+        filename = "temp_upload_#{Time.zone.now.to_i}.pdf"
+        temp_file = Tempfile.new(filename, encoding: 'ASCII-8BIT')
+        temp_file.write(decoded_data)
+        temp_file.close
+        ActionDispatch::Http::UploadedFile.new(filename: filename,
+                                               type: 'application/pdf',
+                                               tempfile: temp_file)
```

Comment:
do we need to worry about cleaning up these temp files?

Reply:
The ActionDispatch::Http::UploadedFile does eventually hit a .close state when the controller finishes the process the same way it would if it were just an uploaded binary file which basically closes out the temp file as well. https://api.rubyonrails.org/v6.0.2.1/classes/ActionDispatch/Http/UploadedFile.html#method-i-close

Reply:
i'm thinking more of an `unlink` (which will delete the file, rather than just close it)

Reply:
oh nm.
>Uploaded files are temporary files whose lifespan is one request. When the object is finalized Ruby unlinks the file, so there is no need to clean them with a separate maintenance task.

Reply:
:+1: